src/java.base/share/classes/jdk/internal/foreign/AbstractMemorySegmentImpl.java

```diff
@@ -141,15 +141,14 @@ public final MemorySegment reinterpret(Arena arena, Consumer<MemorySegment> clea
                 MemorySessionImpl.toMemorySession(arena), cleanup);
     }
 
-    public MemorySegment reinterpretInternal(Class<?> callerClass, long newSize, Scope scope, Consumer<MemorySegment> cleanup) {
+    private NativeMemorySegmentImpl reinterpretInternal(Class<?> callerClass, long newSize, MemorySessionImpl scope, Consumer<MemorySegment> cleanup) {
```

Comment:
This method is also probably missing a `ForceInline`

Reply:
I think it's the other `reinterpret` methods that use `Reflection.getCallerClass` which are alsocritical to force-inline. Since the non-inlined/non-intrinsic version does a very slow stack walk to get the caller.

Reply:
Plus, this method can be expanded to a very large method (before constant folding) so we will need to be careful applying `@ForceInline` to it.